From be902b19ed0c87fa71442d248f05b68337b02eb2 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Thu, 23 Feb 2012 17:16:15 +0800
Subject: [PATCH 23/43] ENGR00175257 Return the dns for ppp connection

framework will directly get the dns from datacall list
instead of to fetch it through the property net.ppp0.dns

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 reference-ril/reference-ril.c | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/reference-ril/reference-ril.c b/reference-ril/reference-ril.c
index 672f43e..e0ac717 100755
--- a/reference-ril/reference-ril.c
+++ b/reference-ril/reference-ril.c
@@ -14,7 +14,7 @@
 ** See the License for the specific language governing permissions and
 ** limitations under the License.
 */
-/* Copyright (C) 2011 Freescale Semiconductor,Inc. */
+/* Copyright (C) 2011-2012 Freescale Semiconductor, Inc. */
 
 #include <telephony/ril_cdma_sms.h>
 #include <telephony/librilutils.h>
@@ -745,7 +745,20 @@ static void requestOrSendDataCallList(RIL_Token *t)
                 /* I don't know where we are, so use the public Google DNS
                  * servers by default and no gateway.
                  */
-                responses[i].dnses = "8.8.8.8 8.8.4.4";
+                //responses[i].dnses = "8.8.8.8 8.8.4.4";
+                //responses[i].gateways = "";
+                const char* separator = " ";
+                const int   dnslist_sz = 128;
+                char*       dnslist = alloca(dnslist_sz);
+                char  propName[PROP_NAME_MAX];
+                memset(dnslist, 0, 128);
+                property_get("net.ppp0.dns1", propName, "8.8.8.8");
+                strlcat(dnslist, propValue, dnslist_sz);
+                strlcat(dnslist, separator, dnslist_sz);
+                property_get("net.ppp0.dns2", propName, "8.8.4.4");
+                strlcat(dnslist, propValue, dnslist_sz);
+
+                responses[i].dnses = dnslist;
                 responses[i].gateways = "";
             }
         }
-- 
1.8.0

